core/core_integrations/messenger.js

```diff
@@ -0,0 +1,198 @@
+        var spl = message.split('\n');
+
+        for (var i = 0; i < spl.length; i++) {
+            var strs = [];
+            while (spl[i].length > 0) {
+                var str = spl[i].substr(0, 320);
+                var ind = str.lastIndexOf(' ');
+                if (ind >= 0 && str.length !== spl[i].length) {
+                    str = str.substr(0, Math.min(str.length, ind + 1));
+                }
+
+                strs.push(str);
+                spl[i] = spl[i].substr(str.length);
+            }
+            spl.splice(i, 1);
+            for (var j = 0; j < strs.length; j++) {
+                spl.splice(i + j, 0, strs[j]);
+            }
+            i += strs.length;
+        }
+
+        for (var i = 0; i < spl.length; i++) {
+            var curr = spl[i];
+            for (var j = i + 1; j < spl.length; j++) {
+                if (curr + spl[j] + 1 < 320) {
+                    curr += '\n' + spl[j];
+                    spl.splice(j, 1);
+                }
+                else {
+                    break;
+                }
+            }
+            spl[i] = curr;
+        }
+
+        return spl;
+    },
+
+    genericMessageRequest = function(message, recipientId, callback) {
+        request({
+            url: 'https://graph.facebook.com/v2.6/me/messages',
+            qs: {access_token: exports.config.token},
+            method: 'POST',
+            json: {
+                recipient: {id: recipientId},
+                message: message
+            }
+        },
+        function(error) {
+            if (error) {
+                throw error;
+            }
+            if (callback) {
+                callback();
+            }
+        });
+    },
+
+    listGenericMessageRequest = function(list, property, threadId) {
+        if (!list || list.length <= 0) {
+            return;
+        }
+        var arg = {};
+        arg[property] = list[0];
+        list.splice(0, 1);
+        genericMessageRequest(arg, threadId, listGenericMessageRequest.bind(this, list, property, threadId));
+    };
+
+exports.start = function(callback) {
+    api = shim.createPlatformModule({
+        sendMessage: function (message, threadId) {
+            var spl = sentenceSplitter(message);
+            listGenericMessageRequest(spl, 'text', threadId);
+        },
+        getUsers: function() {
+            return exports.config.users || {};
+        },
+        sendImage: function(type, image, description, thread) {
+            if (description) {
+                this.sendMessage(description, thread);
+            }
+            switch (type) {
+                case 'url':
+                    genericMessageRequest({
+                        attachment: {
+                            type: 'image',
+                            payload: {
+                                url: image
+                            }
+                        }
+                    }, thread);
+                    break;
+                case 'file':
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected indentation of 12 space characters but found 16.](https://www.codacy.com/app/theprojectshop/Kassy/file/2575844897/issues/source?bid=3282987&fileBranchId=3288129#l131)